### PR TITLE
Remove selection from MaskSel

### PR DESCRIPTION
**Pull request by sritchie**
I realized that `MaskSel` doesn't need to extend another selection. ~all of our current choicemap masking is currently done with a single `MaskSel`... this change tidies up the print representation, offering a light simplification from

```python
In [3]: switch(1, chms)
Out[3]:
XorChm(
  c1=FilteredChm(
    c=StaticChm(c=ValueChm(v=1.0), addr='x'),
    selection=MaskSel(
      s=AllSel(),
      flag=Flag(f=<jax.Array(False, dtype=bool)>),
    ),
  ),
  c2=FilteredChm(
    c=StaticChm(c=ValueChm(v=2.0), addr='y'),
    selection=MaskSel(
      s=AllSel(),
      flag=Flag(f=<jax.Array(True, dtype=bool)>),
    ),
  ),
)
```

to

```python
In [3]: switch(1, chms)
Out[3]:
XorChm(
  c1=FilteredChm(
    c=StaticChm(c=ValueChm(v=1.0), addr='x'),
    selection=MaskSel(flag=Flag(f=<jax.Array(False, dtype=bool)>)),
  ),
  c2=FilteredChm(
    c=StaticChm(c=ValueChm(v=2.0), addr='y'),
    selection=MaskSel(flag=Flag(f=<jax.Array(True, dtype=bool)>)),
  ),
)
```

### CODE REVIEW

**Comment by femtomc** [comment]
love it but it is unsafe

**Comment by sritchie** [comment]
@femtomc fixed on `main`, counting that comment as a +1
